src/sidebar/test/service-url-test.js

```diff
@@ -1,20 +1,166 @@
 'use strict';
 
-var serviceUrl = require('../service-url');
+var proxyquire = require('proxyquire');
 
-describe('serviceUrl', function () {
-  var service;
+/** Return a fake annotationUI object. */
+function fakeAnnotationUI() {
+  var links = null;
+  return {
+    updateLinks: function(newLinks) {
+      links = newLinks;
+    },
+    getState: function() {
+      return {links: links};
+    },
+  };
+}
 
-  beforeEach(function () {
-    service = serviceUrl({serviceUrl: 'https://test.hypothes.is/'});
+function createServiceUrl(linksPromise) {
+  var replaceURLParams = sinon.stub().returns(
+    {url: 'EXPANDED_URL', params: {}}
+  );
+
+  var serviceUrlFactory = proxyquire('../service-url', {
+    './util/url-util': { replaceURLParams: replaceURLParams },
+  });
+
+  var annotationUI = fakeAnnotationUI();
+
+  var features = { flagEnabled: sinon.stub().returns(true) };
+
+  var settings = {};
+
+  var store = {
+    links: sinon.stub().returns(linksPromise),
+  };
+
+  return {
+    annotationUI: annotationUI,
+    store: store,
+    serviceUrl: serviceUrlFactory(annotationUI, features, settings, store),
+    replaceURLParams: replaceURLParams,
+  };
+}
+
+describe('links', function () {
+  context('before the API response has been received', function() {
+    beforeEach(function() {
+      // Create a serviceUrl function with an unresolved Promise that will
+      // never be resolved - it never receives the links from store.links().
+      var parts = createServiceUrl(new Promise(function() {}));
+
+      this.serviceUrl = parts.serviceUrl;
```

Comment:
> I saw a Stack Overflow answer somewhere that said using this was the "right" way to do this with mocha

I can imagine someone saying that. The Mocha docs do make use of `this`. However my objection to use of `this` outside of class constructors/methods in JS is that it isn't obvious what the _type_ of `this` (ie. the constructor and set of valid properties) is.